build.sbt

```diff
@@ -50,15 +56,37 @@ def extraOptions(scalaVersion: String) =
     case _ => Seq.empty
   }
 
+lazy val scala2settings = Seq(
+  libraryDependencies ++= (CrossVersion.partialVersion(scalaVersion.value) match {
+    case Some((2, _)) =>
+      Seq(
+        "org.scala-lang" % "scala-reflect" % scalaVersion.value,
+        compilerPlugin("org.typelevel" %% "kind-projector" % "0.10.3")
+      )
+    case _ => Seq.empty
+  })
+)
+lazy val macroSettings = Seq(
+  libraryDependencies ++= (CrossVersion.partialVersion(scalaVersion.value) match {
+    case Some((2, _)) =>
+      Seq(
+        "com.softwaremill.magnolia1_2" %% "magnolia" % V.magnolia
+      )
+    case _ =>
+      Seq(
+        "com.softwaremill.magnolia1_3" %% "magnolia" % V.magnoliaFor3
+      )
+  })
+)
 val commonSettings = Seq(
   organization := "org.scanamo",
   organizationName := "Scanamo",
   startYear := Some(2019),
   homepage := Some(url("http://www.scanamo.org/")),
   licenses += ("Apache-2.0", new URL("https://www.apache.org/licenses/LICENSE-2.0.txt")),
+  scalaVersion := "2.12.16",
```

Comment:
nitpick: this can be `V.scala212` right?

Reply:
I edited it myself.

Reply:
Thank you! I overlooked it.